src/App.tsx

```diff
@@ -1,66 +1,116 @@
+
+  if (typeof chrome !== "undefined" && chrome.devtools && chrome.devtools.network) {
+    networkAPI = chrome.devtools.network;
+  } else if (typeof browser !== "undefined" && browser.devtools && browser.devtools.network) {
+    networkAPI = browser.devtools.network;
+  } else {
+    throw new Error("Neither Chrome nor Firefox devtools network APIs available");
+  }
+
+  // Wrap getHAR so it returns a Promise
+  networkAPI.getHARAsync = () => {
+    return new Promise((resolve, reject) => {
+      try {
+        networkAPI.getHAR((harLog: any) => resolve(harLog.log || harLog));
+      } catch (err) {
+        reject(err);
+      }
+    });
+  };
+
+  return networkAPI;
+};
 
 type SetAction = {
-  action: "set",
-  requests: Request[]
-}
+  action: "set";
+  requests: Request[];
+};
 
 type IncrementAction = {
-  action: "increment",
-  request: Request
-}
+  action: "increment";
+  request: Request;
+};
 
-type Action = SetAction | IncrementAction
+type Action = SetAction | IncrementAction;
 
-function reducer(requests : Request[], action: Action) {
+function reducer(requests: Request[], action: Action) {
   switch (action.action) {
     case "set":
-      return action.requests
+      return action.requests;
     case "increment":
-      return [...requests, action.request]
+      return [...requests, action.request];
+    default:
+      return requests;
   }
 }
 
 function App() {
+  const [requests, dispatch] = useReducer(reducer, []);
+  const [selectedRequest, selectRequest] = useState<Request | null>(null);
 
-  const [requests, dispatch] = useReducer(reducer, [])
-  const [selectedRequest, selectRequest] = useState<Request | null>(null)
   useEffect(() => {
-    let ignore = false
-    dispatch({action: "set", requests: []})
-    chrome.devtools.network.getHAR((harLog) => {
-      if (!ignore) {
-        dispatch({ action: "set", requests: harLog.entries})
-      }
-    })
-    return () => {
-      ignore = true
+    let ignore = false;
+    dispatch({ action: "set", requests: [] });
+
+    try {
+      const networkAPI = getNetworkAPI();
+      networkAPI
+        .getHARAsync()
+        .then((harLog: any) => {
+          if (!ignore && harLog && harLog.entries) {
+            dispatch({ action: "set", requests: harLog.entries });
+          }
+        })
+        .catch((error: any) => {
+          console.error("Failed to get HAR log:", error);
+        });
+    } catch (error) {
+      console.error("Failed to get network API:", error);
     }
-  }, [])
-  
-  useEffect(() => {
-    const listener = (request : chrome.devtools.network.Request) => { dispatch({action: "increment", request}) }
-    chrome.devtools.network.onRequestFinished.addListener(listener)
+
     return () => {
-      chrome.devtools.network.onRequestFinished.removeListener(listener)
+      ignore = true;
+    };
+  }, []);
+
+  useEffect(() => {
+    try {
+      const networkAPI = getNetworkAPI();
+      const listener = (request: any) => {
```

Comment:
please don't use `any` - I am going to set up a linter that prohibits this very shortly. If the type needs to be wider than `chrome.devtools.network.Request` please use a union